src/app/pages/Send/form/EvmForm.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { FC, useCallback, useMemo } from 'react';
+
+import BigNumber from 'bignumber.js';
+import { isString } from 'lodash';
+import { FormProvider, useForm } from 'react-hook-form-v7';
+import { formatEther, isAddress } from 'viem';
+
+import { DeadEndBoundaryError } from 'app/ErrorBoundary';
+import { useEvmTokenMetadataSelector } from 'app/store/evm/tokens-metadata/selectors';
+import { useFormAnalytics } from 'lib/analytics';
+import { EVM_TOKEN_SLUG } from 'lib/assets/defaults';
+import { useEvmTokenBalance } from 'lib/balances/hooks';
+import { useAssetFiatCurrencyPrice } from 'lib/fiat-currency';
+import { t, toLocalFixed } from 'lib/i18n';
+import { getAssetSymbol } from 'lib/metadata';
+import { useSafeState } from 'lib/ui/hooks';
+import { isEvmNativeTokenSlug } from 'lib/utils/evm.utils';
+import { ZERO } from 'lib/utils/numbers';
+import { getAccountAddressForEvm } from 'temple/accounts';
+import { useAccountForEvm, useVisibleAccounts } from 'temple/front';
+import { useEvmChainByChainId } from 'temple/front/chains';
+import { useEvmAddressByDomainName } from 'temple/front/evm/ens';
+import { useSettings } from 'temple/front/ready';
+
+import { useEvmEstimationData } from '../hooks/use-evm-estimation-data';
+
+import { BaseForm } from './BaseForm';
+import { ReviewData, SendFormData } from './interfaces';
+import { getMaxAmountFiat } from './utils';
+
+interface Props {
+  chainId: number;
+  assetSlug: string;
+  onSelectAssetClick: EmptyFn;
+  onReview: (data: ReviewData) => void;
+}
+
+export const EvmForm: FC<Props> = ({ chainId, assetSlug, onSelectAssetClick, onReview }) => {
+  const account = useAccountForEvm();
+  const network = useEvmChainByChainId(chainId);
+
+  if (!account || !network) throw new DeadEndBoundaryError();
+
+  const allAccounts = useVisibleAccounts();
+  const { contacts } = useSettings();
+
+  const accountPkh = account.address as HexString;
+
+  const formAnalytics = useFormAnalytics('SendForm');
+
+  const { value: balance = ZERO } = useEvmTokenBalance(assetSlug, accountPkh, network);
+  const { value: ethBalance = ZERO } = useEvmTokenBalance(EVM_TOKEN_SLUG, accountPkh, network);
+
+  const [shouldUseFiat, setShouldUseFiat] = useSafeState(false);
+
+  const storedMetadata = useEvmTokenMetadataSelector(network.chainId, assetSlug);
+  const assetMetadata = isEvmNativeTokenSlug(assetSlug) ? network?.currency : storedMetadata;
+
+  const assetDecimals = assetMetadata?.decimals ?? 0;
```

Comment:
Now if assetMetadata not found, the user will see the form that couldn't be submitted (error toast is shown on submit). But what you are proposing means the user will see an error boundary instead of form in this case. Do I understand correctly?